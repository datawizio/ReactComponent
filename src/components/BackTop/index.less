@import "~antd/es/back-top/style/index";
@import "../../styles/variables";

.ant-back-top.dw-back-top {
  bottom: 30px;
  right: 30px;
  .ant-back-top-content {
    background: @primary-color;
    opacity: 0.7;
    &:hover {
      opacity: 1;
    }
    .ant-back-top-icon {
      background: none;
      margin: 8px auto;
    }
  }
}
