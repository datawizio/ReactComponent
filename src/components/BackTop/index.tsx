import React from "react";
import { BackTop as AntBackTop } from "antd";
import { UpOutlined } from "@ant-design/icons";
import "./index.less";

const BackTop = (props) => {
  return (
    <AntBackTop className="dw-back-top" {...props}>
      <div className="ant-back-top-content">
        <div className="ant-back-top-icon">
          <UpOutlined />
        </div>
      </div>
    </AntBackTop>
  );
};

export default BackTop;
