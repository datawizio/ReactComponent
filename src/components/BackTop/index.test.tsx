import "jsdom-global/register";
import React from "react";
import { mount } from "enzyme";

import BackTop from "./index";

const mockProps = {};

const setUp = (props?) => mount(<BackTop {...props} />);

describe("BackTop component", () => {
  let component;
  beforeEach(() => {
    component = setUp(mockProps);
  });

  it("Render BackTop correctly", () => {
    expect(component).toMatchSnapshot();
  });

  it("BackTop has correct class", () => {
    expect(component.find(".dw-back-top").length).toBeTruthy();
  });
});
