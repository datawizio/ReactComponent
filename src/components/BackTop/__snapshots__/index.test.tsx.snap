// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackTop component Render BackTop correctly 1`] = `
<BackTop>
  <BackTop
    className="dw-back-top"
  >
    <div
      className="ant-back-top dw-back-top"
      onClick={[Function]}
    >
      <CSSMotion
        motionName="fade"
        removeOnLeave={true}
        visible={false}
      >
        <DomWrapper />
      </CSSMotion>
    </div>
  </BackTop>
</BackTop>
`;
